src/util/jwt_utils.js

```diff
@@ -16,8 +16,8 @@ function get_jwt_secret() {
 }
 
 function make_auth_token(object = {}, jwt_options = {}) {
-    // Remote services/endpoints should not sign tokens
-    if (config.NOOBAA_AUTH_TOKEN) return config.NOOBAA_AUTH_TOKEN;
+    // Remote services/endpoints should not sign tokens - unless required for STS
+    if (config.NOOBAA_AUTH_TOKEN && object.assumed_role_access_key === undefined) return config.NOOBAA_AUTH_TOKEN;
```

Comment:
@guymguym you had an idea for a fix that includes separating the functions, would you please give more details about it?
Anyway, I think this comment about "Remote services/endpoints" is not updated.